docs/lessons/random.rst

```diff
@@ -0,0 +1,88 @@
+Random
+------
+
+Sometimes you want to leave things to chance, or mix it up a little. You want
+the device to act randomly.
+
+MicroPython comes with a ``random`` module to make it easy to introduce chance
+and a little chaos into your code. For example, here's how to scroll a random
+name across the display::
+
+    from microbit import *
+    import random
+
+    names = ["Amy", "Ben", "Damien", "Alia", "Kushal", "Sophie", "Zoltan" ]
+
+    display.scroll(random.choice(names))
+
+The list (``names``) contains seven names defined as strings of characters.
+Using the "onion" effect introduced when discussing *nesting* the
+``random.choice`` method takes this list as an argument and returns an item
+chosen at random. This item (the randomly chosen name) is the argument for
+``display.scroll``.
+
+Random Numbers
+++++++++++++++
+
+Random numbers are very useful. They're common in games. Why else do we have
+dice?
+
+MicroPython comes with several useful random number methods. Here's how to
+make a simple dice::
+
+    from microbit import *
+    import random
+
+    display.show(str(random.randint(1, 6)))
+
+Every time the device is reset it displays a number between 1 and 6. You're
+starting to get familiar with *nesting* and it's important to note that
+``random.randint`` returns a whole number (also called an integer - hence the
+name of the method) between the two arguments (inclusive). Notice that because
+``display.show`` expects a character then we use the ``str`` function to turn
+the numeric value into a character (we turn, for example, ``6`` into ``"6"``).
+
+If you know you'll always want a number between ``0`` and ``N`` then use the
+``random.randrange`` method. If you give it a single argument it'll return
+random integers up to, but not including, the value of the argument (this is
+different to the behaviour of ``random.randint``).
+
+Sometimes you need numbers with a decimal point in them. These are called
+*floating point* numbers and it's possible to such a number with the
+``random.random`` method. This only returns values between ``0.0`` and ``1.0``.
```

Comment:
Maybe a good idea to specify if this version of random includes the values 0.0 and 1.0 themselves.